onmt/translate/translator.py

```diff
@@ -618,17 +618,24 @@ def _align_pad_prediction(self, predictions, bos, pad):
         )  # (batch, n_best, tgt_l)
         return batched_nbest_predict
 
-    def _report_score(self, name, score_total, words_total):
-        if words_total == 0:
-            msg = "%s No words predicted" % (name,)
+    def _report_score(self, name, score_total, nb_sentences):
+        # In the case of length_penalty = none we report the total logprobs
+        # divided by the number of sentence to get an approximation of the
+        # per sentence logprob. We also return the corresponding ppl
+        # When a length_penalty is used eg: "avg" or "wu" since logprobs
+        # are normalized per token we report the per line per token logprob
+        # and the corresponding "per word perplexity"
+        if nb_sentences == 0:
+            msg = "%s No translations" % (name,)
         else:
-            avg_score = score_total / words_total
-            ppl = np.exp(-score_total.item() / words_total)
-            msg = "%s AVG SCORE: %.4f, %s PPL: %.4f" % (
+            score = score_total / nb_sentences
+            ppl = np.exp(-score_total.item() / nb_sentences)
+            msg = "%s SCORE: %.4f, %s PPL: %.2f NB SENTENCES: %d" % (
```

Comment:
Thanks for the change.
Can we just add a comment that the ppl might contain a length penalty term, even if it was already the case before.

I think you forget the file with the length_penalty and alpha

Reply:
> I think you forget the file with the length_penalty and alpha

another PR is better.



> Can we just add a comment that the ppl might contain a length penalty term, even if it was already the case before.

I think the comments in the code are good enough.

Reply:
sorry, I missed the comments